server/src/routers/LocationsRouter/processLocations.js

```diff
@@ -1,16 +1,43 @@
+import firestore from "../../config/firebase/firebaseConfig";
 import getOneLocation from "./getOneLocation";
 
 import locations from "../../config/locations";
-import { E_NO_API_RES } from "../../config/constants";
+import * as constants from "../../config/constants";
+
+/**
+ *
+ * Processes the locations in the query, calling the getOneLocation 
+ * for a single call, or for multiple calls (in a loop).
+ *
+ * @param {Object} data [the data from yale api axios response]
+ * @param {Object} query [the query passed in by user of our api]
+ *
+ * Returns the location object with the location information for 
+ * one or all locations based on the query
+ * 
+ */
 
 export default async function processLocations(data, query) {
     // throw on bad response from Yale Dining
     if (!data || !data.DATA || !data.DATA.length || !data.DATA[0].length) {
-        throw new Error(E_NO_API_RES);
+        throw new Error(constants.E_NO_API_RES);
+    }
+
+    // Get firebase doc, checking for errors
+    let hoursDoc = undefined;
+    try{
+        hoursDoc = await firestore.doc("locations/hours").get();
+    } catch (e) {
+        throw new Error(constants.E_DB_READ);
+    }
+
+    if (!hoursDoc.exists) {
+        throw new Error(constants.E_DB_NOENT + "locations/hours");
```

Comment:
Do we really want to throw and abandon all hope of returning hours if we can't get to our cache?